src/Behat/Behat/Output/Node/Printer/JUnit/JUnitScenarioPrinter.php

```diff
@@ -76,12 +76,63 @@ public function printOpenTag(Formatter $formatter, FeatureNode $feature, Scenari
         /** @var JUnitOutputPrinter $outputPrinter */
         $outputPrinter = $formatter->getOutputPrinter();
 
-        $outputPrinter->addTestcase(array(
-            'name' => $name,
+        $outputPrinter->addTestcase($this->addCircleCiAttributes(array(
+            'name' => $this->addLineNumberToScenarioName($name, $scenario->getLine()),
+            'file' => $this->addLineNumberToFeatureFile($this->convertToRelativePath($feature->getFile()), $scenario->getLine()),
             'classname' => $feature->getTitle(),
             'status' => $this->resultConverter->convertResultToString($result),
             'time' => $this->durationListener ? $this->durationListener->getDuration($scenario) : ''
-        ));
+        )));
+    }
+
+    /**
+     * Append scenario line number in the feature file to a scenario name
+     *
+     * @param string $name
+     * @param integer $lineNumber
+     * @return string
+     */
+    private function addLineNumberToScenarioName(string $name, int $lineNumber) {
+        return $name . " Line #" . $lineNumber;
+    }
+
+    /**
+     * Append scenario line number in the feature file to the feature file
+     *
+     * This can be used to reference the scenario when running tests in the command line
+     *
+     * @param string $path
+     * @param integer $lineNumber
+     * @return string
+     */
+    private function addLineNumberToFeatureFile(string $path, int $lineNumber) {
+        return $path . ":" . $lineNumber;
+    }
+    /**
+     * Converts the absolute feature file path to the relative path to the root of the project
+     *
+     * @param string $absolutePath
+     * @return string
+     */
+    private function convertToRelativePath(string $absolutePath) {
+        return str_replace(getcwd() . '/', '', $absolutePath);
+    }
+
+    /**
+     * Adds attributes to testcase tag related to CircleCI if in a CircleCI environment
+     *
+     * @param array $attributes
+     * @return array
+     */
+    private function addCircleCiAttributes(array $attributes) {
+        $circleCiNode = getenv('CIRCLE_NODE_INDEX');
+        if($circleCiNode !== false) {
+            $circleCiNodeLabel =  "[Node #$circleCiNode] ";
+            $attributes['name'] = $circleCiNodeLabel . $attributes['name'];
```

Comment:
The `$circleCiNodeLabel` variable seems unnecessary:

```suggestion
            $attributes['name'] = "[Node #$circleCiNode] ${attributes['name']}";
```